packages/suite/src/reducers/wallet/coinjoinReducer.ts

```diff
@@ -665,6 +666,27 @@ export const selectCurrentCoinjoinBalanceBreakdown = (state: CoinjoinRootState)
     return balanceBreakdown;
 };
 
+const memoizeCoinjoinPrison = memoizeWithArgs(
+    (
+        prison: NonNullable<CoinjoinAccount['prison']>,
+        session: CoinjoinAccount['session'],
+        transactionCandidates: CoinjoinAccount['transactionCandidates'],
+    ) => {
+        return Object.keys(prison).reduce<typeof prison>((result, key) => {
+            const inmate = prison[key];
+            // select **only** inmates with assigned roundId (signed in current round or promised to future blaming round)
+            if (
+                inmate.roundId &&
+                (session || transactionCandidates?.some(tx => tx.roundId === inmate.roundId))
+            ) {
+                result[key] = inmate;
+            }
+
+            return result;
+        }, {});
+    },
+);
+
 export const selectRegisteredUtxosByAccountKey = (
```

Comment:
why not memoizing the whole `selectRegisteredUtxosByAccountKey` selector? 
```suggestion
export const selectRegisteredUtxosByAccountKey = memoizeWithArgs(...
```

Reply:
Because I'm extremely conservative and I didn't know how clever `proxy-memoize` is. I'll change it, test it and push a fixup.